glslc/src/file_compiler.cc

```diff
@@ -27,6 +30,32 @@
 namespace {
 using shaderc_util::string_piece;
 
+// A helper function to emit SPIR-V binary code as a list of hex numbers in
+// text form. Returns true if the compilation result is emitted successfully.
+template <typename CompilationResultType>
+bool EmitSpirvBinaryAsCommaSeparatedNumbers(
+    const CompilationResultType& result, std::ostream* out) {
+  // Return early if the compilation output is not in SPIR-V binary code form.
+  if (!std::is_same<CompilationResultType,
+                    shaderc::SpvCompilationResult>::value)
+    return false;
+  // Return early if the compilation result is empty.
+  if (result.begin() == result.end()) return false;
```

Comment:
Since the arg is const, should probably use cbegin() and cend() consistently.
It's just an accident that begin and end return const iterators here.